src/SELF_DGModel3D_t.f90

```diff
@@ -335,9 +332,9 @@ subroutine Write_DGModel3D_t(this,fileName)
 
     print*,"Writing pickup file : "//trim(pickupFile)
 
-    if(this%decomp%mpiEnabled) then
+    if(this%mesh%decomp%mpiEnabled) then
 
-      call Open_HDF5(pickupFile,H5F_ACC_TRUNC_F,fileId,this%decomp%mpiComm)
+      call Open_HDF5(pickupFile,H5F_ACC_TRUNC_F,fileId,this%mesh%decomp%mpiComm)
```

Comment:
Same as with the dgmodel2d comments - we need parallel advection diffusion 3d test cases to test drive parallel writes.